core/domain/rte_component_registry.py

```diff
@@ -64,3 +67,64 @@ def get_tag_list_with_attrs(cls):
                 for ca_spec in component_specs['customization_arg_specs']]
 
         return component_tags
+
+    @classmethod
+    def get_component_types_to_component_classes(cls):
+        """Get component classes mapping for component types."""
+        module_fullname = 'extensions.rich_text_components.components'
+        module_obj = pkgutil.find_loader(module_fullname).load_module(
+            'components')
+        component_types_to_component_classes = {}
+        component_names = cls.get_all_rte_components().keys()
+        for component_name in component_names:
+            for name, obj in inspect.getmembers(module_obj):
+                if inspect.isclass(obj) and name == component_name:
+                    component_types_to_component_classes[
+                        'oppia-noninteractive-%s' % component_name.lower()] = (
+                            obj)
+
+        return component_types_to_component_classes
+
+    @classmethod
+    def get_inline_components(cls):
+        """Gets an list of inline component tags."""
+        rich_text_components_specs = cls.get_all_rte_components()
```

Comment:
There's a lot of repetition in these four methods. When you see that, look for a chance to remove duplication.

Probably having an internal method along the lines of _get_component_tag_names(key, expected_value) might not be a bad idea.

Reply:
Yes, makes the code much better now.